crates/miden-objects/src/account/component/template/storage/entry_content.rs

```diff
@@ -534,15 +534,11 @@ impl MapRepresentation {
             })
             .collect::<Result<Vec<(Digest, Word)>, _>>()?;
 
-        // Validate that no key appears multiple times.
-        let mut seen_keys = BTreeSet::new();
-        for (map_key, _) in entries.iter() {
-            if !seen_keys.insert(map_key) {
-                return Err(AccountComponentTemplateError::StorageMapHasDuplicateKeys(*map_key));
-            }
-        }
-
-        Ok(StorageMap::with_entries(entries))
+            StorageMap::with_entries(entries).map_err(|err| {
+                // Include the error directly instead of using it as a source error, since this
+                // error variant is used in another place as well.
+                AccountComponentTemplateError::StorageMapHasDuplicateKeys(err.to_string().as_bytes().try_into().expect("Unable to convert error to Digest"))
+            })
```

Comment:
Ah, I didn't realize the inner type had changed to `Digest` now.

I think in that case I would go with this approach:

```rust
#[error("storage map contains duplicate keys")]
StorageMapHasDuplicateKeys(#[source] Box<dyn Error + Send + Sync + 'static>),
```

and then here:

```rust
AccountComponentTemplateError::StorageMapHasDuplicateKeys(Box::new(err))
```

and in the other place it is used it becomes:

```rust
if !seen_keys.insert(key) {
    return Err(AccountComponentTemplateError::StorageMapHasDuplicateKeys(
        Box::from(format!("key `{key}` is duplicated")),
    ));
}
```